File: app/src/main/java/com/tichuguru/db/GameDao.java
package com.tichuguru.db;

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import java.util.List;

@Dao
public interface GameDao {
    @NonNull
    @Query("SELECT * FROM games ORDER BY id")
    List<GameEntity> getAll();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(@NonNull GameEntity game);
}

File: app/src/main/kotlin/com/tichuguru/db/GameDao.kt
package com.tichuguru.db

import androidx.room.Dao
import androidx.room.Query
import androidx.room.Upsert

@Dao
interface GameDao {
    @Query("SELECT * FROM games ORDER BY id")
    fun getAll(): List<GameEntity>

    @Upsert
    fun upsert(game: GameEntity): Long

    @Query("DELETE FROM games WHERE id = :id")
    fun deleteById(id: Long)
}
